packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -1111,6 +1154,136 @@ describe('Select', () => {
             await waitForUpdatesAsync();
             expect(pageObject.getDisplayText()).toBe('One one');
         });
+
+        it('programmatically setting selected of current selected option to false results in placeholder being displayed', async () => {
+            await clickAndWaitForOpen(element);
+            pageObject.clickOption(1);
+            const selectedOption = pageObject.getSelectedOption();
+            selectedOption!.selected = false;
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getDisplayText()).toBe('One');
+        });
+    });
+
+    describe('clearable', () => {
+        let element: Select;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+        let pageObject: SelectPageObject;
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setup());
+            element.style.width = '200px';
+            element.clearable = true;
+            await connect();
+            await waitForUpdatesAsync();
+            pageObject = new SelectPageObject(element);
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        it('clear button is visible after selecting an option', async () => {
+            await clickAndWaitForOpen(element);
+            pageObject.clickOption(1);
+            await waitForUpdatesAsync();
+
+            expect(pageObject.isClearButtonVisible()).toBeTrue();
+        });
+
+        it('clicking clear button does not open dropdown', async () => {
+            pageObject.clickClearButton();
+            await waitForUpdatesAsync();
+            expect(element.open).toBeFalse();
+        });
+
+        it('clicking clear button fires change event', () => {
+            const changeEvent = jasmine.createSpy();
+            element.addEventListener('change', changeEvent);
+            pageObject.clickClearButton();
+            expect(changeEvent).toHaveBeenCalledTimes(1);
+        });
+
+        it('if dropdown open clicking clear button closes dropdown and fires change event', async () => {
+            pageObject.clickSelect();
+            await waitForUpdatesAsync();
+            const changeEvent = jasmine.createSpy();
+            element.addEventListener('change', changeEvent);
+            pageObject.clickClearButton();
+            expect(changeEvent).toHaveBeenCalledTimes(1);
+        });
+
+        describe('without placeholder', () => {
+            it('clear button is visible by default', () => {
```

Comment:
This test _does_ depend on not having a placeholder, since if there is, the clear button would not be visible (see test in following `descibe` block).

Reply:
No, I mean this is a test where you're asserting that when the selected option is not a placeholder, the clear button is visible. Whether or not a placeholder option exists doesn't affect the test. The only reason the test in the other `describe` block is asserting differently is because you change the selected option _into_ a placeholder.

Reply:
Updated.